### PR TITLE
feat: start adding the ui component engine

### PR DESCRIPTION
**Pull request by moldy530**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR updates dependencies, config settings, and adds Tailwind CSS components and utilities to the UI demo project.

### Detailed summary
- Updated ESLint config to extend `next/core-web-vitals`
- Updated TypeScript config to include JSX setting
- Added PostCSS config for Tailwind CSS
- Modified build script to ignore `ui-demo`
- Added Tailwind CSS global styles in `globals.css`
- Updated Next.js metadata in `layout.tsx`
- Added Tailwind CSS config in `tailwind.config.ts`
- Added Tailwind CSS components and utilities for forms, buttons, modals, and borders
- Updated TypeScript config for UI demo project
- Updated package.json scripts and dependencies for UI demo
- Added Tailwind CSS config in `tailwind.config.ts`
- Added Tailwind CSS color variables and theme settings

> The following files were skipped due to too many changes: `packages/alchemy/src/tailwind/theme.ts`, `packages/alchemy/src/tailwind/types.ts`, `examples/ui-demo/README.md`, `examples/ui-demo/public/next.svg`, `packages/alchemy/src/tailwind/components/input.ts`, `packages/alchemy/src/tailwind/utils.ts`, `packages/alchemy/src/react/components/button.tsx`, `packages/alchemy/src/react/logos/google.tsx`, `packages/alchemy/src/tailwind/utils.test.ts`, `examples/ui-demo/src/app/AuthModal.tsx`, `examples/ui-demo/src/app/page.tsx`, `packages/alchemy/src/tailwind/plugin.ts`, `yarn.lock`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by moldy530** [comment]
image


^ rendering of the demo set.

I probably won't merge this yet, gonna keep building on top of it until I have a good base for this stuff

**Review by dphilipson — DISMISSED** [review]
This looks great! Really happy with this setup.

File: packages/alchemy/src/tailwind/components/colorsvars.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import { getColorVariableName } from "../utils.js";
+
+export const colorVariables = (theme: AccountKitTheme): ComponentDef => {
+  // This assumes that the colors object is a flat object of type { [name: string]: ColorVariantRecord }
+  const { colors } = theme;
+  const rules = Object.entries(colors).reduce((accum, [name_, variant]) => {
+    const name = name_ as AccountKitThemeColor;
+
+    return {
+      ...accum,
```
[the remark this answers is not in the document]
Comment (moldy530) [reply packages/alchemy/src/tailwind/components/colorsvars.ts:15]: yea totally fair, I do this in another PR so lemme change that. I was being super lazy lol

File: packages/alchemy/src/tailwind/utils.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ * @returns the deeply merged object of S and T
+ */
+export const apply = <S extends Record<string, any>>(
+  source: S,
+  target?: DeepPartial<S>
+): S => {
+  if (!target) return source;
+
+  for (const key in source) {
+    if (!target[key]) {
```
[the remark this answers is not in the document]
Comment (moldy530) [reply packages/alchemy/src/tailwind/utils.ts:36]: ah yea good call, will fix

**Pull request merged**